src/controller/ExampleOperationalCredentialsIssuer.cpp

```diff
@@ -89,6 +89,42 @@ CHIP_ERROR ExampleOperationalCredentialsIssuer::Initialize(PersistentStorageDele
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR ExampleOperationalCredentialsIssuer::GenerateNOCChainAfterValidation(NodeId nodeId, FabricId fabricId,
+                                                                                const Crypto::P256PublicKey & pubkey,
+                                                                                MutableByteSpan & rcac, MutableByteSpan & icac,
+                                                                                MutableByteSpan & noc)
+{
+    ChipLogProgress(Controller, "Generating NOC");
+    X509CertRequestParams noc_request = { 1, mIntermediateIssuerId, mNow, mNow + mValidity, true, fabricId, true, nodeId };
+    ReturnErrorOnFailure(
+        NewNodeOperationalX509Cert(noc_request, CertificateIssuerLevel::kIssuerIsIntermediateCA, pubkey, mIntermediateIssuer, noc));
+
+    ChipLogProgress(Controller, "Generating ICAC");
+    X509CertRequestParams icac_request = { 0, mIssuerId, mNow, mNow + mValidity, true, fabricId, false, 0 };
+    ReturnErrorOnFailure(NewICAX509Cert(icac_request, mIntermediateIssuerId, mIntermediateIssuer.Pubkey(), mIssuer, icac));
+
+    uint16_t rcacBufLen = static_cast<uint16_t>(std::min(rcac.size(), static_cast<size_t>(UINT16_MAX)));
```

Comment:
Why this std::min with a UINT16_MAX?

Reply:
Because we need to pass the length to an API that takes `uint16_t`....  That said, we could probably just fail if `rcac.size()` somehow does not fit in `uint16_t` and cast directly....

Reply:
I think its ok if the rcac size is > UINT16_MAX. We will not use all the buffer, as RCAC is guaranteed to fit in much smaller space than UINT16_MAX. We reduce the size of Span after filling in the RCAC.